src/Common/src/CoreLib/Interop/Windows/Kernel32/Interop.FormatMessage.cs

```diff
@@ -38,50 +39,51 @@ internal static string GetMessage(int errorCode)
 
         internal static string GetMessage(IntPtr moduleHandle, int errorCode)
         {
-            var sb = new StringBuilder(InitialBufferSize);
+            Span<char> buffer = stackalloc char[InitialBufferSize];
             do
             {
-                string errorMsg;
-                if (TryGetErrorMessage(moduleHandle, errorCode, sb, out errorMsg))
+                if (TryGetErrorMessage(moduleHandle, errorCode, buffer, out string errorMsg))
                 {
                     return errorMsg;
                 }
-                else
-                {
-                    // increase the capacity of the StringBuilder.
-                    sb.Capacity *= BufferSizeIncreaseFactor;
-                }
+
+                // Increase the capacity of the buffer.
+                buffer = new char[buffer.Length * BufferSizeIncreaseFactor];
```

Comment:
Array pool?

Reply:
Rare case on error path. Not worth it.

Reply:
If we wanted to do something about this, using `FORMAT_MESSAGE_ALLOCATE_BUFFER` and avoiding the retry logic completely would be best.